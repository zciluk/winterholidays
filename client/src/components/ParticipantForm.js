import React from "react";
import { Field, reduxForm } from "redux-form";
import moment from "moment";
import momentLocalizer from "react-widgets-moment";
import DateTimePicker from "react-widgets/lib/DateTimePicker";
import "react-widgets/dist/css/react-widgets.css";
import { SubmissionError } from "redux-form";
import { connect } from "react-redux";
import { sendForm } from "../actions";
// Form controller
moment.locale("pl");
momentLocalizer();

//validations
const required = value => (value ? undefined : "Required");
const email = value =>
  value && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)
    ? "Invalid email address"
    : undefined;

// Render function DTPicker
const renderDateTimePicker = ({
  input: { onChange, value },
  placeholder,
  meta: { touched, error, visited }
}) => {
  const className = `field ${error && touched ? "error" : ""}`;
  return (
    <div className={className}>
      <DateTimePicker
        autoOk={true}
        onChange={onChange}
        format="DD/MMM/YYYY"
        time={false}
        value={value ? new Date(value) : null}
        placeholder={placeholder}
      />
      {(touched || visited) &&
        (error && <div className="ui pointing red basic label">{error}</div>)}
    </div>
  );
};

// Render for other input fields (name, surname, e-mail)
const renderField = ({ input, meta: { touched, error }, placeholder }) => {
  const className = `field ${error && touched ? "error" : ""}`;
  return (
    <div className={className}>
      <div>
        <input {...input} placeholder={placeholder} />
        {touched &&
          (error && <div className="ui pointing red basic label">{error}</div>)}
      </div>
    </div>
  );
};
//Main class with render-submit functions
class ParticipantForm extends React.Component {
  onSubmit = formValues => {
    const error = "Error saving form! Try again.";

    if (
      !formValues.email ||
      !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(formValues.email)
    ) {
      throw new SubmissionError({ _error: error });
    } else if (!formValues.name) {
      throw new SubmissionError({ _error: error });
    } else if (!formValues.surname) {
      throw new SubmissionError({ _error: error });
    } else if (!formValues.datePicker) {
      throw new SubmissionError({ _error: error });
    } else {
      this.props.sendForm(formValues);
    }
  };

  render() {
    return (
      <form
        onSubmit={this.props.handleSubmit(this.onSubmit)}
        className="ui form error"
      >
        <Field
          id="name"
          name="name"
          component={renderField}
          placeholder="Name"
          validate={required}
        />
        <Field
          id="surname"
          name="surname"
          component={renderField}
          placeholder="Surname"
          validate={required}
        />
        <Field
          id="email"
          name="email"
          component={renderField}
          placeholder="E-mail"
          validate={[email, required]}
        />
        <Field
          id="datePicker"
          name="datePicker"
          component={renderDateTimePicker}
          showTime={false}
          placeholder="Date of holiday"
          validate={required}
        />
        <button id="submit" className="ui teal button">Submit</button>
        {this.props.error && <strong>{this.props.error}</strong>}
      </form>
    );
  }
}

const formWrapped = reduxForm({
  form: "ParticipantForm"
})(ParticipantForm);

export default connect(
  null,
  { sendForm }
)(formWrapped);
