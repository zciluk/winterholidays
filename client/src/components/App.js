import React from 'react';
import ParticipantForm from './ParticipantForm'
// main container component
const App = () => {
    const containerStyle = {
        margin: '3rem'
      };
    return(
        <div className="ui text container" style={containerStyle} >
        
            <div className="ui segment"> 
                <div className="ui teal ribbon label">
                    <i className="big snowflake icon"></i> X-mas holiday form!
                </div>
                <ParticipantForm />
             </div>
        </div>
    );

};

export default App;