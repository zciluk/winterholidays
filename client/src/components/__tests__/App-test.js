import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import ParticipantForm from '../ParticipantForm';
// basic shallow render checks
describe('App component', () => {
  test('should shallow correctly', () => {
      expect(shallow(
        <App />
      )).toMatchSnapshot() 
  })
  it('renders ParticipantForm component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<ParticipantForm />)).toBe(true);
  })  
})