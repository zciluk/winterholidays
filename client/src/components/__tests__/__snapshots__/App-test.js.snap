// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should shallow correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ui segment"
      >
        <div
          className="ui teal ribbon label"
        >
          <i
            className="big snowflake icon"
          />
           X-mas holiday form!
        </div>
        <Connect(Component) />
      </div>,
      "className": "ui text container",
      "style": Object {
        "margin": "3rem",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ui teal ribbon label"
          >
            <i
              className="big snowflake icon"
            />
             X-mas holiday form!
          </div>,
          <Connect(Component) />,
        ],
        "className": "ui segment",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="big snowflake icon"
              />,
              " X-mas holiday form!",
            ],
            "className": "ui teal ribbon label",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "big snowflake icon",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " X-mas holiday form!",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ui segment"
        >
          <div
            className="ui teal ribbon label"
          >
            <i
              className="big snowflake icon"
            />
             X-mas holiday form!
          </div>
          <Connect(Component) />
        </div>,
        "className": "ui text container",
        "style": Object {
          "margin": "3rem",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ui teal ribbon label"
            >
              <i
                className="big snowflake icon"
              />
               X-mas holiday form!
            </div>,
            <Connect(Component) />,
          ],
          "className": "ui segment",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="big snowflake icon"
                />,
                " X-mas holiday form!",
              ],
              "className": "ui teal ribbon label",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "big snowflake icon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " X-mas holiday form!",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
